test/test_e2e.rb

```diff
@@ -104,4 +104,14 @@ def test_spec_file_splitting
       "./spec/fast_spec.rb",
     ], queue)
   end
+
+  def test_suite_with_failures_and_fail_fast
+    queue = exec_build("failing_suite", "--fail-fast 1")
+
+    assert queue.build_failed_fast?
+    refute queue.build_successful?
+    assert queue.example_failures.keys.length +
+           queue.non_example_errors.keys.length >=
+           queue.fail_fast
+  end
```

Comment:
Also assert that `queue.fail_fast == 1`. I know it's implicitly tested by the above assertion, but we should better be explicit in the tests.

Reply:
We also need to assert that there are even more jobs at the queue, that won't get processed because we failed fast (i.e. `key_queue_unprocessed`).